libr/socket/run.c

```diff
@@ -218,46 +227,63 @@ static char *getstr(const char *src, R_NULLABLE size_t *out_len) {
 				}
 			}
 			// slurp file
-			ret = r_file_slurp (src + 1, &len);
+			ret = r_file_slurp (src, &len);
+			break;
 		}
 	case '`':
 		{
-		char *msg = strdup (src + 1);
-		int msg_len = strlen (msg);
-		if (msg_len > 0) {
-			msg [msg_len - 1] = 0;
-			char *ret = r_sys_cmd_str (msg, NULL, NULL);
-			r_str_trim_tail (ret);
-			free (msg);
-			len = strlen (ret);
-			goto beach;
-		}
+		int msg_len = strlen (src);
+		if (msg_len == 0) {
+			R_LOG_ERROR ("Invalid backtick expression in input");
+			return NULL;
+		}
+		if (src [msg_len-1] != '`') {
+			R_LOG_ERROR ("Unterminated backtick expr in input");
+			return NULL;
+		}
+		char *msg = strdup (src);
+		if (!msg) {
+			return NULL;
+		}
+		msg [msg_len-1] = 0;
+		int cmd_len;
+		ret = r_sys_cmd_str (msg, NULL, &cmd_len);
```

Comment:
check if `!ret`. cmd_len seems to be always initialized by r_sys_cmd_str(). so i think thats not a problem

Reply:
Will change to initialize `cmd_len = 0`. But IMO `ret` can be NULL since it's not dereferenced after the break